source/develop/release-management/features/network/ipv6_subnets.html.md

```diff
@@ -0,0 +1,179 @@
+With IPv6 IPAM in place, routing amongst different IPv6 subnets should be
+possible, and achieved through an OVN logical router.
+
+## Objectives
+
+Provide guests (virtual machines) attached to OVN networks IPv6 address support.
+The following configurations, which are explaining in the [IPAM types](#ipam-types) section, should be possible:
+
+* static addresses
+* stateful DHCPv6
+* stateless DHCPv6
+* stateless address auto-configuration
+
+This feature will focus on providing support for IPv6 static addresses, and stateful DHCPv6. This means that stateless configuration, from router advertisement messages, is out of scope at this time.
+
+## IPAM
+
+### IPAM types
+
+IPv6 provides the following IPAM options:
+
+* slaac: neighbor discovery is provided through router advertisement messages - *RA*s. Both the IP address and the connection paramenters (such as hop limit, MTU, etc) are configured from these router advertisement messages. Specified in [SLAAC rfc](https://tools.ietf.org/html/rfc4862).
+
+* dhcpv6-stateless: IP addresses are assigned by OVN's IPv6 stack, based on the port's mac address and the advertised prefix - in the RA messages.The connection parameters (MTU, max hops, DNS, etc) are configured through the subnet's native DHCPv6 service.
+
+* dhcpv6-stateful: Both the IP addresses and the connection parameters are extraced from the router RA messages.
+
+### IPAM configuration requirements
+
+* slaac: this IPAM configuration does not rely on OVN's subnet DHCP service, which means a DHCP_Options object is **not** required. The RA messages are sent by an OVN Logical_Router, thus one is obviously required.
+
+* dhcpv6-stateless: this IPAM configuration relies on OVN's subnet DHCP services to get its IP address, and on the Logical_Router's RA message's connection parameters to properly configure the connection. Thus, a DHCP_Options entry is required, as is a Logical_Router object.
+
+* dhcpv6-stateful: this IPAM configuration relies solely in OVN's subnet DHCP services, and is very close to the IPv4 configuration counterpart - a DHCP_Options object is required, having both an IPv6 cidr, and ip_version set to IPv6.
+
+### IPAM configuration
+
+This section explains how to configure each of the IPAM options. They are listed by development priority.
+
+* dhcpv6-stateful
+  - create an ovn DHCP_Options object, having an ipv6 cidr configured.
+    It requires the user to specify the ethernet address for the DHCPserver
+    to use in the *options:server_id* DHCP_Options attribute.
+  - to enable IPAM for IPv6, the logical switch (ovn entity) where the ports are
+    created has to include the **other_config: ipv6_prefix** key defined.
+    The value should be the the IP prefix featured in the provisioned cidr
+    parameter, ending in '::' - e.g. 8230:5678::, for instance.
+
+* slaac
+  - configure the logical router port(s) to operate in *slaac* mode.
+    To do that, the logical router port *ipv6_ra_configs:address_mode* key
```

Comment:
maybe <address_mode>?

Reply:
Nope, that's the key we need to set to slaac - we're not implementing much here; just configuring OVN to work properly.

And that's done by setting the **address_mode** parameter, in the **ipv6_ra_configs** dict within the logical router port object to **slaac**.

Reply:
Agree